### PR TITLE
Metrics overrides for result and duration

### PR DESCRIPTION
**Pull request by awschristou**
## Types of changes

- [x] Infrastructure (change which improves the lifecycle management (CI/CD, build, package, deploy, lint, etc) of the application, but does not change functionality.)

## Description

Two changes to the telemetry system:
* Added constants for standard metadata field names and values
* `result` and `duration` will now retain existing metadata values if they exist

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/telemetry/telemetryEvent.ts [hunk 1]
```diff
@@ -10,6 +10,14 @@ import { MetadataEntry, MetricDatum, Unit } from './clienttelemetry'
 const NAME_ILLEGAL_CHARS_REGEX = new RegExp('[^\\w+-.:]', 'g')
 const REMOVE_UNDERSCORES_REGEX = new RegExp('_', 'g')
 
+export const METADATA_FIELD_NAME = {
+    RESULT: 'result',
+    DURATION: 'duration',
+}
+
+export const METADATA_RESULT_PASS = 'Succeeded'
```
Comment (mpiroc) [inline src/shared/telemetry/telemetryEvent.ts]: Would it make sense for this to be an enum?

File: src/shared/telemetry/telemetryUtils.ts [hunk 2]
```diff
@@ -85,3 +88,9 @@ export function registerCommand<T>({
         thisArg
     )
 }
+
+function setMetadataIfNotExists(metadata: Map<string, string>, key: string, value: string) {
+    if (!metadata.has(key)) {
```
Comment (mpiroc) [inline src/shared/telemetry/telemetryUtils.ts]: `Map.set` is idempotent, so the only case where this test would matter is if it was previously set to a *different* value. In this case, should we throw?

Comment (awschristou) [reply src/shared/telemetry/telemetryUtils.ts]: No, we wish to use the existing value if there is one

File: src/shared/telemetry/telemetryEvent.ts (hunk 1 above)
Comment (awschristou) [reply src/shared/telemetry/telemetryEvent.ts]: I considered enum, but ultimately we want to emit a string, so I opted for the simple approach

**Pull request merged**